### PR TITLE
updates to decrypt encrypted context attributes from the generators i…

### PR DESCRIPTION
**Pull request by shriramshankar**
…n the heartbeat response

### CODE REVIEW

File: src/heartbeat/utils.js [hunk 1]
```diff
@@ -39,9 +44,13 @@ function updateCollectorConfig(collectorConfig) {
...
  * @param {Object} ctx - The context from the generator
  * @param {Object} def - The contextDefinition from the generator template
+ * @param {Object} refocusInstance - The refocus instance object, contaning the
+ * instance name, the refocus collector token for this instance and the refocus
+ * instance url
+ * @param {Object} res - The heartbeat response object
  * @returns {Object} the context object with default values populated
  */
-function assignContextDefaults(ctx, def) {
+function assignContext(ctx, def, refocusInstance, res) {
```
Comment (iamigo) [inline src/heartbeat/utils.js]: there are no tests for a falsey "def" arg

**Comment by shriramshankar** [comment]
@iamigo @pallavi2209 Updated the PR.

File: src/heartbeat/utils.js [hunk 2]
```diff
@@ -50,10 +59,19 @@ function assignContextDefaults(ctx, def) {
...
+  const secret = collectorToken + heartbeatTimestamp;
+
   Object.keys(def).forEach((key) => {
     if (!ctx.hasOwnProperty(key) && def[key].hasOwnProperty('default')) {
       ctx[key] = def[key].default;
     }
+
+    if (ctx.hasOwnProperty(key) && def.hasOwnProperty(key) &&
+      def[key].encrypted) {
+      ctx[key] = commonUtils.decrypt(ctx[key], secret, encryptionAlgorithm);
```
Comment (pallavi2209) [inline src/heartbeat/utils.js]: Do we need to handle the case in which decrypt function fails for some reason (eg, incorrect algo) ? @iamigo

Comment (iamigo) [reply src/heartbeat/utils.js]: Can you please create a user story to deal with that?

**Pull request merged**